include/boost/http_proto/serializer.hpp

```diff
@@ -297,44 +325,130 @@ class BOOST_SYMBOL_VISIBLE
+    stream& operator= (
+        stream const& other) = default;
 
+    /**
+        A MutableBufferSequence consisting of a buffer pair.
+     */
     using buffers_type =
         buffers::mutable_buffer_pair;
 
+    /**
+        Returns the remaining available capacity.
+
+        The returned value represents the available free
+        space in the backing fixed-sized buffers used by the
+        serializer associated with this stream.
+
+        The capacity is absolute and does not do any
+        accounting for any octets required by a chunked
+        transfer encoding.
+    */
     BOOST_HTTP_PROTO_DECL
     std::size_t
     capacity() const noexcept;
 
+    /**
+        Returns the number of octets serialized by this
+        stream.
+
+        The associated serializer stores stream output in its
+        internal buffers. The stream returns the size of this
+        output.
+    */
     BOOST_HTTP_PROTO_DECL
     std::size_t
     size() const noexcept;
 
+    /**
+        Return true if the stream cannot currently hold
+        additional output data.
+
+        The fixed-sized buffers maintained by the associated
+        serializer can be sufficiently full from previous
+        calls to @ref stream::commit.
+
+        This function can be called to determine if the user
+        should drain the serializer via @ref serializer::consume calls
+        before attempting to fill the buffer sequence
+        returned from @ref stream::prepare.
+    */
     BOOST_HTTP_PROTO_DECL
     bool
     is_full() const noexcept;
 
+    /**
+        Returns a MutableBufferSequence for storing
+        serializer input. If `n` bytes are written to the
+        buffer sequence, @ref stream::commit must be called
+        with `n` to update the backing serializer's buffers.
+
+        The returned buffer sequence is as wide as is
+        possible.
+
+        @exception std::length_error Thrown if the stream
+        has insufficient capacity and a chunked transfer
+        encoding is being used
+    */
     BOOST_HTTP_PROTO_DECL
     buffers_type
     prepare() const;
 
+    /**
+        Serialize and commit `n` bytes.
+
+        Once the sequence returned from @ref prepare has been
+        filled, the input can be serialized and committed to the
+        associated serializer's output area via a call to `commit(n)`.
+
+        If a chunked transfer encoding is being used then commit
+        is responsible for writing the chunk-header and also the
+        closing CRLF for the chunk-data. `n` denotes the size
+        of the chunk.
+
+        @exception std::logic_error Thrown if commit is
+        called with 0. Instead, the closing chunk must be
+        written by a call to @ref stream::close.
+    */
     BOOST_HTTP_PROTO_DECL
     void
     commit(std::size_t n) const;
 
+    /**
+        Close the stream.
+
+        close() writes the last-chunk to the underlying buffers
+        of the stream's associated serializer, i.e. `0\r\n\r\n`.
```

Comment:
`stream` can be used even when there is no chunked transfer-encoding. It is not a good idea to promise implementation details. I do like that this sort of thing is documented but it needs to be done in the implementation as regular non-extracted comments rather than javadocs on the declaration.

It is better to explain why the user needs to call the function rather than to explain what it does internally. For example:

"This function should be called to indicate that no more data representing body octets will be provided."